Update multi-image3 to catch failures in multi-image generation

nicebert opened the pull request:
Update multi-image testing to catch if all images were actually created. Also explicitly checks for target features xnack+ and xnack-.

nicebert committed:
Update multi-image3 to catch failures in multi-image generation

Update multi-image testing to catch if all images were actually created. Also explicitly checks for target features xnack+ and xnack-.

jplehr commented on test/smoke/multi-image3/Makefile:
```diff
@@ -15,4 +15,4 @@ CC           = $(OMP_BIN) $(VERBOSE)
 include ../Makefile.rules
 run: $(TESTNAME)
```
The `run` rule is not used by the `check_smoke` script, so, this extra part will not be tested.

Please use the `RUNCMD` to provide more complex rules than `./$(TESTNAME)`. The `RUNCMD` will be used by the `check` rule when set.

nicebert replied:
made the change as requested :)

nicebert committed:
Update Makefile

Update Makefile to make use of RUNCMD as suggested by JP so that check smoke script catches the issue

nicebert committed:
Update Makefile

Removes explicit target features from offload-arch specification (causes error when AOMP_GPU matches one of the explicitly specified targets with different features)

jplehr commented on test/smoke/multi-image3/Makefile:
```diff
@@ -5,14 +5,13 @@ TESTSRC_MAIN = multi-image.c
-OMP_FLAGS    = -fopenmp --offload-arch=$(AOMP_GPU)$(AOMP_TARGET_FEATURES) --offload-arch=gfx803$(AOMP_TARGET_FEATURES) --offload-arch=gfx1031$(AOMP_TARGET_FEATURES)
+RUNCMD	     = ./$(TESTNAME) && strings $(TESTNAME) | grep $(AOMP_GPU)$(AOMP_TARGET_FEATURES) && strings $(TESTNAME) | grep gfx908$(AOMP_TARGET_FEATURES) && strings $(TESTNAME) | grep gfx942$(AOMP_TARGET_FEATURES)
+
+OMP_FLAGS    = -fopenmp --offload-arch=$(AOMP_GPU)$(AOMP_TARGET_FEATURES),gfx908$(AOMP_TARGET_FEATURES),gfx942$(AOMP_TARGET_FEATURES)
```
Question: Is this set somewhere outside (or is this a usecase that should be supported)?
If not, then I think it would be better to remove it, to not lead to confusion.

nicebert replied:
I'm not quite sure but as it was set in the original test (before my commits modifying it adding xnack as explicit target feature)

jplehr reviewed (commented):
One final question, other than that: I tested locally and LGTM

jplehr reviewed (approved):
LGTM
